### PR TITLE
feat(parameters): query & path parameters now accept x-go-name

### PR DESCRIPTION
**Pull request by maxatome**
Related to #2167

### CODE REVIEW

File: generator/templates/client/parameter.gotmpl [hunk 1]
```diff
@@ -91,7 +91,7 @@ type {{ pascalize .Name }}Params struct {
   {{ blockcomment .Description }}
 
   {{ end }}*/
-  {{ pascalize .ID }} {{ if and (not .IsArray) (not .IsMap) (not .HasDiscriminator) (not .IsInterface) (not .IsStream) (or .IsNullable  ) }}*{{ end }}{{ if not .IsFileParam }}{{ .GoType }}{{ else }}runtime.NamedReadCloser{{end}}
+  {{ or (index .Extensions "x-go-name") (pascalize .ID) }} {{ if and (not .IsArray) (not .IsMap) (not .HasDiscriminator) (not .IsInterface) (not .IsStream) (or .IsNullable  ) }}*{{ end }}{{ if not .IsFileParam }}{{ .GoType }}{{ else }}runtime.NamedReadCloser{{end}}
```
Comment (fredbi) [inline generator/templates/client/parameter.gotmpl]: @maxatome, there is also a templates/server/parameter.gotmpl

Overall, I am not convinced of the template-based approach to alias parameters. Have you thought about changing it here instead? https://github.com/go-swagger/go-swagger/blob/1c98855b472d8782c366459428ec2a5e8339ccf2/generator/operation.go#L741

Comment (maxatome) [reply generator/templates/client/parameter.gotmpl]: You are right it is a way simpler :)
Done.

**Comment by fredbi** [comment]
@maxatome did you test it with client generation, I mean does your spec produce some code that builds from both client and server? Hint: you can add this to our CI spec build here: https://github.com/go-swagger/go-swagger/blob/master/hack/codegen-fixtures.yaml (note: since you have a "bad spec", you should specify only the one that is supposed to work, with its full path).

**Comment by maxatome** [comment]
@fredbi yes it works for client generation too.
Note I added unit tests to check that case as well as server one.

**Comment by fredbi** [comment]
I've seen the unit test, I am talking about making sure (manually, I am not asking for more automated testing here) that the client compiles properly.

**Comment by maxatome** [comment]
In the `client/xxx_parameters.go` files, if we have:
- `name: foo`
- `x-go-name: bar`

We have the methods `WithBar(foo string)` and `SetBar(foo string)`.

**Pull request merged**